### PR TITLE
Add the side panel to the folders to enable or disable them

### PR DESCRIPTION
**Pull request by matiasdelellis**
Fix issue #171 

## Enabled..
imagen

## Disabled
imagen

## Not supported..
imagen

### CODE REVIEW

File: js/files-tabview.js [hunk 1]
```diff
@@ -21,99 +22,164 @@
...
+        getFolderTemplate: function (data) {
+            var html = "";
+            html += "<div class='emptycontent'>";
+            html += "<div class='icon-user svg'></div>";
+            html += "<p>";
+            html += "<input class='checkbox' id='searchPersonsToggle'";
+            if (data.child_detection)
+                html += "checked='checked'";
+            html += "type='checkbox'>";
+            html += "<label for='searchPersonsToggle'>" + t('facerecognition', 'Search for my loved ones in this folder.') + "</label>";
```
Comment (stalker314314) [inline js/files-tabview.js]: this "loved ones" seems very patronizing to me (and can be far out of context). I would keep it neutral ("search for faces"/"apply face recognition"/...)

File: js/files-tabview.js [hunk 2]
```diff
@@ -21,99 +22,164 @@
...
+            html += "</p>";
+            html += "<span>" + t('facerecognition', 'Folders with <em>.nomedia</em> files still will be ignored.') + "</span>";
```
Comment (stalker314314) [inline js/files-tabview.js]: Folders with <em>.nomedia</em> file will remain unaffected by this.

File: lib/Service/FileService.php [hunk 3]
```diff
@@ -138,6 +146,32 @@ public function allowsChildDetection(Folder $folder): bool {
 		return true;
 	}
 
+	/**
+	 * Set this folder to enable or disable the analysis using the .facerecognition.json file.
+	 *
+	 * @param Folder $folder Folder to enable/disable for
+	 * @return bool true if the change is done. False if failed.
+	 */
+	public function setAllowChildDetection(Folder $folder, bool $detection): bool {
```
Comment (stalker314314) [inline lib/Service/FileService.php]: Ha ha, maybe change to DescendantDetection or something, this is creepy:)

File: js/files-tabview.js (hunk 2 above)
Comment (stalker314314) [inline js/files-tabview.js]: maybe warn user that new file will be created in this directory?

File: js/files-tabview.js (hunk 1 above)
Comment (matiasdelellis) [reply js/files-tabview.js]: haha.. :sweat_smile: 
I always thought the same .. but I had based on google photos.

Comment (matiasdelellis) [reply js/files-tabview.js]: Maybe .. 'Search for persons in the images of this directory.'

File: js/files-tabview.js (hunk 2 above)
Comment (matiasdelellis) [reply js/files-tabview.js]: MM.. 
I would rather not be so technical.  Do not even notify about .nomedia file.

Maybe: "If the images are outside the gallery they will be ignored."

**Pull request merged**